### PR TITLE
Add improvements to CLI documentation

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Improve inline documentation for the commands
- Improve readme
- Update autogenerated docs - make sure the flags described properly.

**Related issue(s)**
See also #230

### CODE REVIEW

File: docs/gen-docs/kyma_install.md [hunk 1]
```diff
@@ -16,28 +61,28 @@ kyma install [flags]
...
   -h, --help                help for install
-  -n, --noWait              Do not wait for the Kyma installation to complete
-  -o, --override []string   Path to yaml file with parameters to override. Multiple entries of this flag are allowed (default [])
-  -p, --password string     Predefined cluster password
-  -s, --source string       Installation source
-      --src-path string     Path to local sources
+  -n, --noWait              Determines if the command should wait for the Kyma installation to complete.
+  -o, --override []string   Specifies the path to a yaml file with parameters to override. (default [])
+  -p, --password string     Specifies the predefined cluster password.
+  -s, --source string       Specifies the installation source. To use the specific release, write kyma install --source=1.3.0. To use the latest master, write kyma install --source=latest. To use the local sources, write kyma install --source=local. To use the remote image, write kyma install --source=user/my-kyma-installer:v1.4.0
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_install.md]: Cant' use this :( GO forbids. But I will try changing it during the second phase of doc polishing.

File: docs/gen-docs/kyma_install.md [hunk 2]
```diff
@@ -16,28 +61,28 @@ kyma install [flags]
...
+      --tlsKey string       Specifies the TLS key for the domain used for installation.
 ```
 
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_install.md]: already fixed.

File: docs/gen-docs/kyma_install.md [hunk 3]
```diff
@@ -16,28 +61,28 @@ kyma install [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_install.md]: already fixed.

File: docs/gen-docs/kyma_provision.md [hunk 4]
```diff
@@ -15,14 +15,14 @@ Provisions a cluster for installation
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision.md]: already fixed.

File: docs/gen-docs/kyma_provision.md [hunk 5]
```diff
@@ -15,14 +15,14 @@ Provisions a cluster for installation
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision.md]: already fixed.

File: docs/gen-docs/kyma_provision_minikube.md [hunk 6]
```diff
@@ -13,24 +13,24 @@ kyma provision minikube [flags]
...
 ```
-      --cpus string                  CPUs to use (default "4")
-      --disk-size string             Disk size to use (default "30g")
+      --cpus string                  Specifies the number of required CPUs. (default "4")
+      --disk-size string             Specifies the disk size. (default "30g")
   -h, --help                         help for minikube
-      --hypervVirtualSwitch string   Name of the hyperv switch, required if --vm-driver=hyperv
-      --memory string                Memory to use (default "8192")
-      --vm-driver string             VMDriver to use. Possible values: virtualbox,vmwarefusion,kvm,xhyve,hyperv,hyperkit,kvm2,none (default "hyperkit")
+      --hypervVirtualSwitch string   Specifies of the Hyper-V switch if you choose the Hyper-V as the driver.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_minikube.md]: already fixed.

File: docs/gen-docs/kyma_provision_minikube.md [hunk 7]
```diff
@@ -13,24 +13,24 @@ kyma provision minikube [flags]
...
+      --memory string                Specifies the amount of memory. (default "8192")
+      --vm-driver string             Specifies the VMDriver. Possible values: virtualbox,vmwarefusion,kvm,xhyve,hyperv,hyperkit,kvm2,none (default "hyperkit")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_minikube.md]: already fixed.

[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_minikube.md]: already fixed.

File: docs/gen-docs/kyma_provision_minikube.md [hunk 8]
```diff
@@ -13,24 +13,24 @@ kyma provision minikube [flags]
...
 ```
 
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_minikube.md]: already fixed.

File: docs/gen-docs/kyma_provision_minikube.md [hunk 9]
```diff
@@ -13,24 +13,24 @@ kyma provision minikube [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_minikube.md]: already fixed.

File: docs/gen-docs/kyma_test.md [hunk 10]
```diff
@@ -15,18 +15,18 @@ Run tests on a provisioned Kyma cluster
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test.md]: already fixed.

File: docs/gen-docs/kyma_test.md [hunk 11]
```diff
@@ -15,18 +15,18 @@ Run tests on a provisioned Kyma cluster
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test.md]: already fixed.

File: docs/gen-docs/kyma_test_definitions.md [hunk 12]
```diff
@@ -19,13 +19,13 @@ kyma test definitions [flags]
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_definitions.md]: already fixed.

File: docs/gen-docs/kyma_test_definitions.md [hunk 13]
```diff
@@ -19,13 +19,13 @@ kyma test definitions [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_definitions.md]: already fixed.

File: docs/gen-docs/kyma_test_delete.md [hunk 14]
```diff
@@ -21,13 +21,13 @@ kyma test delete <test-suite-1> <test-suite-2> ... <test-suite-N> [flags]
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_delete.md]: already fixed.

File: docs/gen-docs/kyma_test_delete.md [hunk 15]
```diff
@@ -21,13 +21,13 @@ kyma test delete <test-suite-1> <test-suite-2> ... <test-suite-N> [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_delete.md]: already fixed.

File: docs/gen-docs/kyma_test_list.md [hunk 16]
```diff
@@ -19,13 +19,13 @@ kyma test list [flags]
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_list.md]: already fixed.

File: docs/gen-docs/kyma_test_list.md [hunk 17]
```diff
@@ -19,13 +19,13 @@ kyma test list [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_list.md]: already fixed.

File: docs/gen-docs/kyma_test_run.md [hunk 18]
```diff
@@ -30,13 +30,13 @@ kyma test run <test-definition-1> <test-defintion-2> ... <test-definition-N> [fl
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_run.md]: already fixed.

File: docs/gen-docs/kyma_test_run.md [hunk 19]
```diff
@@ -30,13 +30,13 @@ kyma test run <test-definition-1> <test-defintion-2> ... <test-definition-N> [fl
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_run.md]: already fixed.

File: docs/gen-docs/kyma_test_status.md [hunk 20]
```diff
@@ -4,15 +4,17 @@ Shows the status of a test suite and it's related test executions
 
 ### Synopsis
 
-Shows the status of a test suite and it's related test executions
+Use this command to display the status of a test suite and the related test executions.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_status.md]: already fixed.

File: docs/gen-docs/kyma_test_status.md [hunk 21]
```diff
@@ -4,15 +4,17 @@ Shows the status of a test suite and it's related test executions
...
+
+If you don't provide any arguments, the status of all test suites will be printed.
+
+To print status of all test suites, run:
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_status.md]: already fixed.

File: docs/gen-docs/kyma_test_status.md [hunk 22]
```diff
@@ -4,15 +4,17 @@ Shows the status of a test suite and it's related test executions
...
 
-Status of all test suites will be printed if no arguments given
+    kyma test status
 
-#Print status of all test suites
-kyma test status
+To print status of the specific test cases, run:
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_status.md]: already fixed.

File: docs/gen-docs/kyma_test_status.md [hunk 23]
```diff
@@ -28,13 +30,13 @@ kyma test status <test-suite-1> <test-suite-2> ... <test-suite-N> [flags]
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_status.md]: already fixed.

File: docs/gen-docs/kyma_test_status.md [hunk 24]
```diff
@@ -28,13 +30,13 @@ kyma test status <test-suite-1> <test-suite-2> ... <test-suite-N> [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_test_status.md]: already fixed.

File: docs/gen-docs/kyma_uninstall.md [hunk 25]
```diff
@@ -1,16 +1,19 @@
...
 
-Uninstalls Kyma from a running Kubernetes cluster
+Uninstalls Kyma from a running Kubernetes cluster.
 
 ### Synopsis
 
-Uninstalls Kyma from a running Kubernetes cluster.
+Use this command to uninstall Kyma from a running Kubernetes cluster.
+
+Make sure that your KUBECONFIG is already pointing to the target cluster.<br>
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_uninstall.md]: already fixed.

File: docs/gen-docs/kyma_uninstall.md [hunk 26]
```diff
@@ -1,16 +1,19 @@
...
 
-Make sure that your KUBECONFIG is already pointing to the target cluster.
 This command:
-- Removes your cluster administrator account
-- Removes Tiller
-- Removes Kyma Installer
+- Removes your cluster administrator account.
+- Removes Tiller.
+- Removes Kyma Installer.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_uninstall.md]: already fixed.

File: docs/gen-docs/kyma_uninstall.md [hunk 27]
```diff
@@ -21,19 +24,19 @@ kyma uninstall [flags]
...
+      --timeout duration   Time-out after which Kyma CLI stops watching the uninstallation progress. (default 30m0s)
 ```
 
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_uninstall.md]: already fixed.

File: docs/gen-docs/kyma_uninstall.md [hunk 28]
```diff
@@ -21,19 +24,19 @@ kyma uninstall [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_uninstall.md]: already fixed.

File: docs/gen-docs/kyma_version.md [hunk 29]
```diff
@@ -1,10 +1,10 @@
 ## kyma version
 
-Version of the Kyma CLI and connected Kyma cluster
+Displays the version of the Kyma CLI and the connected Kyma cluster.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_version.md]: already fixed.

File: docs/gen-docs/kyma_version.md [hunk 30]
```diff
@@ -1,10 +1,10 @@
...
 
 ### Synopsis
 
-Prints the version of the Kyma CLI and the version of the Kyma cluster the current KUBECONFIG points to. For details, see https://github.com/kyma-project/cli/tree/master/pkg/kyma/docs/version.md.
+Use this command to print the version of the Kyma CLI and the version of the Kyma cluster the current KUBECONFIG points to.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_version.md]: already fixed.

File: docs/gen-docs/kyma_version.md [hunk 31]
```diff
@@ -21,13 +21,13 @@ kyma version [flags]
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the KUBECONFIG file. (default "/Users/d063295/.kube/config")
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_version.md]: already fixed.

File: docs/gen-docs/kyma_version.md [hunk 32]
```diff
@@ -21,13 +21,13 @@ kyma version [flags]
...
+      --non-interactive     Enables the non-interactive shell mode.
+  -v, --verbose             Displays detailed details about the actions triggered by the command.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_version.md]: already fixed.

File: cmd/kyma/provision/minikube/cmd.go [hunk 33]
```diff
@@ -57,17 +57,18 @@ func NewCmd(o *Options) *cobra.Command {
...
 		Aliases: []string{"m"},
 	}
 
-	cmd.Flags().StringVar(&o.VMDriver, "vm-driver", defaultVMDriver, "VMDriver to use. Possible values: "+strings.Join(drivers, ","))
-	cmd.Flags().StringVar(&o.HypervVirtualSwitch, "hypervVirtualSwitch", "", "Name of the hyperv switch, required if --vm-driver=hyperv")
-	cmd.Flags().StringVar(&o.DiskSize, "disk-size", "30g", "Disk size to use")
-	cmd.Flags().StringVar(&o.Memory, "memory", "8192", "Memory to use")
-	cmd.Flags().StringVar(&o.CPUS, "cpus", "4", "CPUs to use")
+	cmd.Flags().StringVar(&o.VMDriver, "vm-driver", defaultVMDriver, "Specifies the VM driver. Possible values: "+strings.Join(drivers, ","))
+	cmd.Flags().StringVar(&o.HypervVirtualSwitch, "hypervVirtualSwitch", "", "Specifies the version of Hyper-V switch if you choose Hyper-V as the driver.")
```
Comment (kazydek) [inline cmd/kyma/provision/minikube/cmd.go]:
```suggestion
	cmd.Flags().StringVar(&o.HypervVirtualSwitch, "hypervVirtualSwitch", "", "Specifies the Hyper-V switch version if you choose Hyper-V as the driver.")
```

File: docs/gen-docs/kyma_uninstall.md [hunk 34]
```diff
@@ -20,20 +23,20 @@ kyma uninstall [flags]
 ### Options
 
 ```
-  -h, --help               help for uninstall
-      --timeout duration   Time-out after which Kyma CLI stops watching the uninstallation progress (default 30m0s)
+      --help               Displays help for the command.
+      --timeout duration   Time-out after which Kyma CLI stops watching the uninstallation progress. (default 30m0s)
```
Comment (kazydek) [inline docs/gen-docs/kyma_uninstall.md]: process of uninstalling?

File: cmd/kyma/uninstall/cmd.go [hunk 35]
```diff
@@ -41,21 +41,24 @@ func NewCmd(o *Options) *cobra.Command {
...
+
+### Usage
 `,
 		RunE:    func(_ *cobra.Command, _ []string) error { return cmd.Run() },
 		Aliases: []string{"i"},
 	}
 
-	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 30*time.Minute, "Time-out after which Kyma CLI stops watching the uninstallation progress")
-
+	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 30*time.Minute, "Time-out after which Kyma CLI stops watching the uninstallation progress.")
```
Comment (kazydek) [inline cmd/kyma/uninstall/cmd.go]: process of uninstalling Kyma?

File: cmd/kyma/provision/minikube/cmd.go (hunk 33 above)
Comment (bszwarc) [reply cmd/kyma/provision/minikube/cmd.go]: Done.

File: cmd/kyma/uninstall/cmd.go (hunk 35 above)
Comment (bszwarc) [reply cmd/kyma/uninstall/cmd.go]: yep, done.

File: docs/gen-docs/kyma_uninstall.md [hunk 36]
```diff
@@ -20,20 +23,19 @@ kyma uninstall [flags]
...
+      --timeout duration   Time-out after which Kyma CLI stops watching the the process of unstalling Kyma. (default 30m0s)
 ```
 
 ### Options inherited from parent commands
 
 ```
-      --kubeconfig string   Path to kubeconfig (default "/Users/i504462/.kube/config")
-      --non-interactive     Do not use spinners
-  -v, --verbose             verbose output
+      --kubeconfig string   Specifies the path to the kubeconfig file. (default "/Users/user/.kube/config")
```
Comment (clebs) [inline docs/gen-docs/kyma_uninstall.md]: This is not true for windows.

The default value is calculated by the kubernetes library as follows: `$HOME_DIR + ".kube/config"`.
To reflect that we can document it something like:
`--kubeconfig string   Specifies the path to the kubeconfig file. (default "$HOME/.kube/config")`

Comment (bszwarc) [reply docs/gen-docs/kyma_uninstall.md]: I agree - changed :)

**Pull request merged**